### PR TITLE
chore: yamllint all the things

### PR DESCRIPTION
**Pull request by dave-tucker**
This adds the same yamllint config that we use in Kepler and adds a Github Workflow to ensure checked in YAML has been correctly linted. Also fixes some minor syntax issues with the GitHub Templates.

### CODE REVIEW

**Comment by sunya-ch** [comment]
Could you please check prometheus-k8s yaml due to the error in this line `The Role "prometheus-k8s" is invalid: rules[0].apiGroups: Required value: resource rules must supply at least one api group`?

File: .github/workflows/build-push.yml [hunk 1]
```diff
@@ -60,66 +57,65 @@ jobs:
...
-        context: .
-        platforms: linux/amd64
-        push: ${{ inputs.push }}
-        tags: ${{ env.image }}
-        file: dockerfiles/Dockerfile
+      - name: Build-push s3 image
+        if: ${{ needs.check-secret.outputs.available == 'true' && env.s3_change == 'true' }}
+        uses: docker/build-push-action@v6
+        with:
+          platforms: llinux/amd64
```
Comment (sthaha) [inline .github/workflows/build-push.yml:97]: typo @dave-tucker

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
-    "python.analysis.extraPaths": [
-        "./src/util"
-    ]
-}
\ No newline at end of file
+  "python.analysis.extraPaths": ["./src/util"],
+  "yaml.schemas": {
+    "https://raw.githubusercontent.com/ansible/ansible-lint/main/src/ansiblelint/schemas/ansible.json#/$defs/tasks": "file:///Users/dave/dev/kepler-model-server/model_training/tekton/tasks/extract-task.yaml"
+  }
+}
```
Comment (sthaha) [inline .vscode/settings.json:6]: How about we delete this file and ignore it? 
Also contains `Users/dave`...

File: manifests/server/kustomization.yaml [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
-- fieldref:
-    fieldPath: spec.template.spec.containers[0].ports[0].containerPort
-  name: MODEL_SERVER_PORT
-  objref:
-    group: apps
-    kind: Deployment
-    name: kepler-model-server
-    version: v1
+  - fieldref:
+      fieldpath: metadata.namespace }
```
Comment (sthaha) [inline manifests/server/kustomization.yaml:8]: `}` added at the end

File: manifests/server/kustomization.yaml [hunk 4]
```diff
@@ -1,29 +1,29 @@
...
+    name: MODEL_SERVER_NAMESPACE
+    objref:
+      group: apps
+      kind: Deployment
+      name: kepler-model-server
+      version: v1
+  - fieldref:
+      fieldpath: spec.template.spec.containers[0].ports[0].containerPort
```
Comment (sthaha) [inline manifests/server/kustomization.yaml:16]: Wonder how this even worked earlier 🤔

File: manifests/server/online-train/patch-trainer.yaml [hunk 5]
```diff
@@ -19,16 +19,16 @@ spec:
...
+          imagePullPolicy: IfNotPresent
+          volumeMounts:
+            - name: cfm
+              mountPath: /etc/kepler/kepler.config
+              readOnly: true
+            - name: mnt
+              mountPath: /mnt
+              readOnly: false
+          command: [python3.10]
+          args: [-u, src/train/online_trainer.py]
```
Comment (sthaha) [inline manifests/server/online-train/patch-trainer.yaml:34]: could you please verify if this change in indentation (scope) does not break anything?

File: manifests/test/model-request-client.yaml [hunk 6]
```diff
@@ -7,18 +7,21 @@ spec:
...
-          name: tmp
+        - name: kepler-exporter
+          image: localhost:5001/kepler_model_server:devel-test
+          imagePullPolicy: IfNotPresent
+          command: [/bin/sh]
+          args:
+            [
+              -c,
+              python3.10 -u tests/weight_model_request_test.py && echo Done && sleep infinity,
+            ]
```
Comment (sthaha) [inline manifests/test/model-request-client.yaml:18]:
```suggestion
          args:
            - -c,
            - python3.10 -u tests/weight_model_request_test.py && echo Done && sleep infinity
```

**Review by sthaha — CHANGES_REQUESTED** [review]

**Comment by sunya-ch** [comment]
Is this duplicated with https://github.com/sustainable-computing-io/kepler-model-server/pull/429?

**Pull request closed without merging**